### PR TITLE
Add `CODEOWNERS` file

### PR DESCRIPTION
**Pull request by sergejparity**
This PR adds `CODEOWNERS` file to the repository.
It is needed to increase awareness of the developers about related code changes.
Currently it adds CI team related stuff. But feel free to propose additions or changes to it.

### CODE REVIEW

**Review by alvicsam — APPROVED** [review]

**Review by HCastano — CHANGES_REQUESTED** [review]
I'm not sure about the value this brings just yet (and the same goes for all the other ink! related CODEOWNER PRs opened today). Will need to discuss this with the team, please don't merge this or the other PRs.

**Comment by sergejparity** [comment]
> @sergejparity I'm not sure about the value this brings just yet (and the same goes for all the other ink! related CODEOWNER PRs opened today). Will need to discuss this with the team, please don't merge this or the other PRs.

**Comment by sergejparity** [comment]
> @sergejparity I'm not sure about the value this brings just yet (and the same goes for all the other ink! related CODEOWNER PRs opened today). Will need to discuss this with the team, please don't merge this or the other PRs.

@HCastano right now we are introducing this feature into all mirrored to GitLab repositories. 
This particular PR intended to bring value first of all for @paritytech/ci team. We'll have early notifications about code changes which touches CI stuff. 
We believe that this might help us more efficiently mitigate risks like security issue with `ink-waterfall`, which happened last week.

But another teams/dev's might benefit from this too.

File: CODEOWNERS
Comment (Robbepop) [inline CODEOWNERS:18]:
```suggestion

# All of the core team members are global code owners.
* @Robbepop @cmichi @ascjones @HCastano 

```

File: CODEOWNERS
Comment (Robbepop) [inline CODEOWNERS:22]:
```suggestion
/.github/ @paritytech/ci @Robbepop @cmichi @ascjones @HCastano
/scripts/.ci/ @paritytech/ci @Robbepop @cmichi @ascjones @HCastano
/.gitlab-ci.yml @paritytech/ci @Robbepop @cmichi @ascjones @HCastano
```

Comment (Robbepop) [reply CODEOWNERS:22]: copied from above

**Review by Robbepop — COMMENTED** [review]
I personally like the idea of code owners although I would have only added the global code owners since there seems to be a conflicting interest between CI team and ink! Core team of who is required to review CI related items. To the interest of the ink! team we'd not introduce others but the global code owners. To the interest of the CI team we'd also introduce the `@paritytech/ci` code owner for the few CI related files. Although I'd like to keep the `/.github/` folder a bit more under the ink! Core team control and only include there what is truly relevant to the CI team.

File: CODEOWNERS
Comment (sergejparity) [reply CODEOWNERS:18]: @Robbepop completely agree with your proposal. We do not want to take over any leverages from ink! team. Our goal is to  improve processes we are responsible for and `CODEOWNERS` feature is a low hanging fruit for this.
Added changes you proposed.

File: CODEOWNERS
Comment (sergejparity) [reply CODEOWNERS]: Added changes you proposed.

File: CODEOWNERS
Comment (HCastano) [inline CODEOWNERS:23]:
```suggestion
/.github/workflows/ @paritytech/ci @Robbepop @cmichi @ascjones @HCastano
```

**Review by HCastano — COMMENTED** [review]
@sergejparity what are the expectations from the CI team going forward? Does the team simply want visibility as to when the CI gets touched, or will be it expected that someone from the CI team approves before we merge anything CI related?

File: CODEOWNERS
Comment (sergejparity) [reply CODEOWNERS]: @HCastano the main point for us is to get automated notifications about changes to CI stuff.

File: CODEOWNERS
Comment (HCastano) [inline CODEOWNERS:16]:
```suggestion
#   you can use them for mentioning unlike an email.
```

**Review by HCastano — APPROVED** [review]
Okay, this is fine with me as long as the CI team doesn't end up blocking the development team from making changes to the CI

**Review by Robbepop — COMMENTED** [review]
I'd like for the `CODEOWNERS` file to exist under the `.github` directory since it kinda is a concept we primarily use by GitHub, or am I wrong?

**Comment by sergejparity** [comment]
> I'd like for the `CODEOWNERS` file to exist under the `.github` directory since it kinda is a concept we primarily use by GitHub, or am I wrong?

This repo as well mirrored into [GitLab](https://gitlab.parity.io/) instance and for sake of unification we'd like to keep it in the root of the repo. It might simplify some our future tasks.

**Pull request merged**
